modules/tinymce/src/plugins/table/test/ts/browser/InlineEditorInsideTableTest.ts

```diff
@@ -35,13 +39,97 @@ describe('browser.tinymce.plugins.table.InlineEditorInsideTableTest', () => {
     inline: true,
     base_url: '/project/tinymce/js/tinymce',
     plugins: 'table',
-    menubar: false,
+    menubar: 'table',
     statusbar: false
   }, setupElement, [ Plugin, Theme ]);
 
+  const focusEditor = (editor: Editor) => {
+    EditorFocus.focus(editor, false);
+  };
+
+  const pWaitForMenuBar = (editor: Editor) => TinyUiActions.pWaitForUi(editor, '.tox-menubar');
+
+  const pAssertMenuButtonDisabled = (editor: Editor, selector: string, disabled: boolean) =>
+    TinyUiActions.pWaitForUi(editor, `div.tox-collection__item[title="${selector}"][aria-disabled="${disabled}"]`);
+
   it('TBA: Table outside of inline editor should not become resizable', () => {
     const editor = hook.editor();
     Mouse.mouseOver(TinyDom.targetElement(editor));
     UiFinder.notExists(SugarBody.body(), 'div[data-row="0"]');
   });
+
+  it('TINY-6625: Table menu items of inline editor shoud be disabled', async () => {
+    const editor = hook.editor();
+    focusEditor(editor);
+    await pWaitForMenuBar(editor);
+    TinyUiActions.clickOnMenu(editor, 'span:contains("Table")');
+    await pAssertMenuButtonDisabled(editor, 'Delete table', true);
+    await pAssertMenuButtonDisabled(editor, 'Table properties', true);
+  });
+
+  context('TINY-6625: Table outside of inline editor should not be affected by table plugin commands', () => {
+    it('TINY-6625: mceTableApplyCellStyle', () => {
+      const editor = hook.editor();
+      focusEditor(editor);
+      editor.execCommand('mceTableApplyCellStyle', false, { 'background-color': 'pink' });
+      const td = editor.getBody().parentElement;
+      assert.notEqual(td.style.backgroundColor, 'pink');
+    });
+
+    it('TINY-6625: mceTableInsertRowBefore', () => {
+      const editor = hook.editor();
+      focusEditor(editor);
+      const beforeCount = count('tr');
+      editor.execCommand('mceTableInsertRowBefore');
+      assert.equal(count('tr'), beforeCount);
```

Comment:
For debugging purposes, it could be better if the count function returned the array of elements and then used `assert.lengthOf` as that means it can print what it found in the error message.